session-loader/Development/IDE/Session.hs

```diff
@@ -380,18 +381,18 @@ data TargetDetails = TargetDetails
   }
 
 fromTargetId :: [FilePath]          -- ^ import paths
+             -> [String]            -- ^ extensions to consider
              -> TargetId
              -> IdeResult HscEnvEq
              -> DependencyInfo
              -> IO [TargetDetails]
 -- For a target module we consider all the import paths
-fromTargetId is (TargetModule mod) env dep = do
+fromTargetId is exts (TargetModule mod) env dep = do
     let fps = [i </> moduleNameSlashes mod -<.> ext | ext <- exts, i <- is ]
-        exts = ["hs", "hs-boot", "lhs"]
```

Comment:
Normally we set optExtensions to be `hs` and `lhs`. Before that, we were looking for `hs-boot` too. Is that behaviour deliberate? Was looking for `hs-boot` wrong? If looking for `hs-boot` wasn't wrong, why wasn't `lhs-boot` on the list?

Reply:
Yes, that's a good point. I think "hs-boot" needs to be added to the list

Reply:
I don't know what's the story with `lhs-boot`. https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/separate_compilation.html does not mention them so perhaps they are not a thinh

Reply:
Fair enough. I wish both boot files AND lhs files were not a thing, so the combination of them is a bit of a monster!